### PR TITLE
RHINENG-761: Handle Compliance tailoring file request

### PR DESCRIPTION
**Pull request by LightOfHeaven1994**
### All Pull Requests:

Check all that apply:

* [x] Have you followed the guidelines in our Contributing document, including the instructions about commit messages?
* [x] Is this PR to correct an issue?
* [x] Is this PR an enhancement?

### Complete Description of Additions/Changes:

**Closes** https://issues.redhat.com/browse/RHINENG-761 
In some situations our customers could  get TCP connection issue and then openscap parser fails to use tailoring file as instead of xml format we get json with error messages. We need better handling of tailoring file request.

### CODE REVIEW

File: insights/client/apps/compliance/__init__.py [hunk 1]
```diff
@@ -109,8 +109,13 @@ def download_tailoring_file(self, profile):
             "https://{0}/compliance/profiles/{1}/tailoring_file".format(self.config.base_url, profile['id'])
         )
         logger.debug("Response code: {0}".format(response.status_code))
-        if response.content is None:
-            logger.info("Problem downloading tailoring file for {0} to {1}".format(profile['attributes']['ref_id'], tailoring_file_path))
+
+        if not response.ok:
+            logger.info("Something went wrong with the downloading tailoring file for {0}. Expected status code 200 but got {1}".format(profile['attributes']['ref_id'], response.status_code))
```
Comment (skateman) [inline insights/client/apps/compliance/__init__.py:114]:
```suggestion
            logger.info("Something went wrong during downloading the tailoring file of {0}. The expected status code is 200, got {1}".format(profile['attributes']['ref_id'], response.status_code))
```

File: insights/client/apps/compliance/__init__.py [hunk 2]
```diff
@@ -109,8 +109,13 @@ def download_tailoring_file(self, profile):
...
+            return None
+
+        if response.content is None or response.headers['Content-Type'] != "application/xml":
+            logger.info("Problem with the content of downloaded tailoring file for {0}. Expected xml but got {1}".format(profile['attributes']['ref_id'], response.headers['Content-Type']))
```
Comment (skateman) [inline insights/client/apps/compliance/__init__.py:118]:
```suggestion
            logger.info("Problem with the content of the downloaded tailoring file of {0}. The expected format is xml, got {1}".format(profile['attributes']['ref_id'], response.headers['Content-Type']))
```

File: insights/client/apps/compliance/__init__.py (hunk 1 above)
Comment (LightOfHeaven1994) [reply insights/client/apps/compliance/__init__.py]: @skateman fixed, thanks

File: insights/client/apps/compliance/__init__.py (hunk 2 above)
Comment (LightOfHeaven1994) [reply insights/client/apps/compliance/__init__.py]: @skateman fixed, thanks

**Review by skateman — APPROVED** [review]
The Seal of Approval

**Review by xiangce — APPROVED** [review]
Looks good to me as well, based on the approval from @skateman

**Comment by xiangce** [comment]
test me

**Comment by LightOfHeaven1994** [comment]
@xiangce I think Insights-QE pipeline fails always for all PRs I saw

**Pull request merged**